### ISSUE TITLE
[RC12, dev] AUTOSENSDATA null: data is old

### ISSUE THREAD

**Issue by chaosbiber** [MilosKozak/AndroidAPS#1490]
### Problem:
Loop stopped working for me this morning without any update or manual data entries because of “old” autosense data for both **OpenAPS SMB** and **AMA** with **any sensitivity plugin**. OpenAPS MA works because it doesn’t use autosense.

In Loop Tab I get “noapsselected” String and in OpenAPS (SMB or AMA) Tab “openaps_noasdata” String - no results

Looking at differnt Logs it seems the autosens data is always 20-30 minutes old, with smb or ama plugin stopping calculations. (Has to be < 11 minutes to be used)
`AUTOSENSDATA null: data is old` (IobCobCalculatorPlugin.java:483) with timestamp more than the 11 minute threshold in the past (~25 minutes)

COB is always N/A, thus Bolus Calculator and Loop won't work.

Time on phone, Nightscout server and pump are in sync.

### Version:
AndroidAPS 2.0h-dev
Build: 126f3925e-2018.10.05-12:01
Flavor: fulldebug
Nightscout-Version: 0.11.0-dev-20180805

While the problem first occured I was on pre-RC12-branch "carb_fix", tried some debugging being on rc12, then switched to dev after resetting the local database and restarting phone.

### Log Highlights:

Here Autosens calculation gets aborted "regularly" mid-calculations (about 20h in the past), interrupted by new BG
```
12:26:57.322 [Thread-375] D/AUTOSENS: [IobCobOref1Thread.run():384]: AutosensData: 04.10.2018 15:59:00 x Delta=-4.0 avgDelta=-2.0 Bgi=-11.864999999999998 Deviation=7.864999999999998 avgDeviation=9.0 Absorbed=0.7542857142857142 CarbsFromBolus=0.0 COB=7.703492857142853 autosensRatio=1.3 slopeFromMaxDeviation=-0.12195121951219512 slopeFromMinDeviation=0.0
12:26:57.322 [Thread-375] D/AUTOSENS: [IobCobOref1Thread.run():121]: Aborting calculation thread (trigger): onEventNewBG
12:26:57.323 [Thread-375] D/AUTOSENS: [IobCobOref1Thread.run():395]: AUTOSENSDATA thread ended: onEventNewBG
```
Then immediately after a new Autosens thread starts and finishes seconds later with about 25 minute old data (data from 12:04, calculated at 12:27):
```
12:27:05.713 [Thread-378] D/AUTOSENS: [IobCobOref1Thread.run():384]: AutosensData: 05.10.2018 12:04:00 x Delta=1.0 avgDelta=0.0 Bgi=-14.518 Deviation=15.518 avgDeviation=14.0 Absorbed=1.4409571428571428 CarbsFromBolus=0.0 COB=30.599380952380955 autosensRatio=1.0 slopeFromMaxDeviation=0.0 slopeFromMinDeviation=1.0714285714285714
12:27:05.715 [Thread-378] D/AUTOSENS: [IobCobOref1Thread.run():395]: AUTOSENSDATA thread ended: onEventNewBG
```
I don't know why he doesn't calculate more recent data, can't find exceptions.
[AndroidAPS.log](https://github.com/MilosKozak/AndroidAPS/files/2450016/AndroidAPS.log)

**Comment by chaosbiber** [comment]
I think I narrowed the problem down a bit.
I believe the problem is triggered by my Libre Transmitter which recently has a differences between each reading of about 5 minutes **and 3-5 seconds**. I charged it yesterday evening, which fits somehow, but I think the time differences have existed before.
This way, when creating the buckets for Autosens, there are some readings missing to fill all of them when enforcing the 5 minute-steps.

Here the most recent bg reading is at 12:26:53 (TZ offset 2h, CEST)
```
12:26:57.436 [Thread-378] D/AUTOSENS: [IobCobCalculatorPlugin.createBucketedData5min():282]: Adding. bgTime: 2018-10-05T10:21:49Z lastbgTime: 2018-10-05T10:26:53Z BgReading{date=1538734909000, date=05.10.2018 12:21:49, value=193.0, direction=null, raw=0.0}
```
After creating the buckets the last one is at 12:03:01
```
12:26:58.008 [Thread-378] D/AUTOSENS: [IobCobCalculatorPlugin.createBucketedData5min():296]: Adjusting bucketed data time. Current: 2018-10-05T10:26:53Z to: 2018-10-05T10:03:01Z by 1432 sec
12:26:58.008 [Thread-378] D/AUTOSENS: [IobCobCalculatorPlugin.createBucketedData5min():301]: Bucketed data created. Size: 344
```
with the result of
```
AUTOSENSDATA null: data is old (OpenAPSPlugin) size()=683 lastdata=05.10.18 12:04
```

During the bucket creating the time diff increases:
```
2:26:57.768 [Thread-378] D/AUTOSENS: [IobCobCalculatorPlugin.createBucketedData5min():296]: Adjusting bucketed data time. Current: 2018-10-04T05:33:05Z to: 2018-10-04T05:33:01Z by 4 sec
12:26:57.769 [Thread-378] D/AUTOSENS: [IobCobCalculatorPlugin.createBucketedData5min():296]: Adjusting bucketed data time. Current: 2018-10-04T05:38:09Z to: 2018-10-04T05:38:01Z by 8 sec
12:26:57.769 [Thread-378] D/AUTOSENS: [IobCobCalculatorPlugin.createBucketedData5min():296]: Adjusting bucketed data time. Current: 2018-10-04T05:43:14Z to: 2018-10-04T05:43:01Z by 13 sec
```
Can we create bigger buckets or would this cause problems during the later calculations? How does this even work when there are sensor gaps?

**Comment by chaosbiber** [comment]
The "adjusted" value, the time gap between bucket and real bg time increases very steady, so I don't know why this problem appeared this morning.
![screenshot 2018-10-05 at 15 22 26](https://user-images.githubusercontent.com/8959896/46538111-68567d80-c8b3-11e8-9e78-2b6acdecac54.png)

**Comment by swissalpine** [comment]
Probably it's related to https://github.com/MilosKozak/AndroidAPS/issues/1395
Today I changed again a sensor and the problem reappears.

**Comment by MilosKozak** [comment]
the problem is that algorithm analyzed data as correct 5 min interval because the time diff was constantly 5 min 4 sec
But in this case is was wrong

**Comment by MilosKozak** [comment]
fixed in 2.0i-dev

**Comment by chaosbiber** [comment]
I can happily confirm the fix works.
Problem reappeared this evening on previous version and updating helped, while simple restarts didn't.

Only remark I could add: when the problem occurred the loop seemed to be running without any indication it would not (apart from glucose values off target and no corrections happening). Maybe additional alarms to notify the user that something's wrong could be added.
